main/src/ca/uwaterloo/flix/api/lsp/CodeActionTriggerKind.scala

```diff
@@ -0,0 +1,27 @@
+package ca.uwaterloo.flix.api.lsp
+
+import org.json4s.{JInt, JValue}
+import ca.uwaterloo.flix.util.Result
+import ca.uwaterloo.flix.util.Result.{Err, Ok}
+
+sealed trait CodeActionTriggerKind
+
+object CodeActionTriggerKind {
+  def parse(json: JValue): Result[CodeActionTriggerKind, String] = {
+    def err(v: JValue) =
+      Err(s"Code action trigger kind not recognized as a valid integer: '$v'.")
+
+    json match {
+      case JInt(i) => i.toInt match {
+        case 1 => Ok(Invoked)
+        case 2 => Ok(Automatic)
+        case _ => err(json)
+      }
+      case _ => err(json)
```

Comment:
I think it would be better with two separate error messages.

Reply:
But more to the point, according to the protocol these unknown values should not really occur. Hence it might be better for us to throw exceptions?

Reply:
I think it's more in line with the rest of the code to treat it as a `Result`

Reply:
In general you are totally write. But in this case its really should not fail because if it does something is very wrong.

In any case, I suspect that if we remove the list of diagnostics then we can remove this parsing code, and it becomes moot.

Reply:
right*